apps/hpc-ftsadmin/src/app/components/flow-form/flow-version.tsx

```diff
@@ -0,0 +1,180 @@
+  bg-unocha-error-light
+  border-unocha-error
+`;
+const MarginEndLink = tw(Link)`
+  me-2
+`;
+const TableRowClick = tw(TableRow)`
+  transition-all
+  hover:shadow-md
+  bg-blend-hue
+  hover:bg-opacity-20
+  hover:cursor-pointer
+`;
+
+const FlowVersions = ({
+  flow,
+  inactiveReasons,
+}: {
+  flow: flows.GetFlowResult;
+  inactiveReasons: categories.Category[];
+}) => {
+  const { lang } = getContext();
+
+  const [selectedVersions, setSelectedVersions] =
+    useState<FlowVersionSelection>([null, null]);
+
+  const hasFlowsSelected = (
+    selection: FlowVersionSelection
+  ): selection is [FlowVersion, FlowVersion] =>
+    selection.every((flowVersion) => flowVersion !== null);
+
+  const pendingReviewCategory = inactiveReasons.find(
+    (category) => category.name === PENDING_REVIEW
+  );
+
+  const handleVersionSelection = (flowVersion: FlowVersion) => {
+    setSelectedVersions((prev) => {
+      const checkedRowIndex = prev.findIndex(
+        (a) => a?.id === flowVersion.id && a.versionID === flowVersion.versionID
+      );
+
+      if (checkedRowIndex !== -1) {
+        const prevSelectedRowsClone = structuredClone(prev);
+        prevSelectedRowsClone[checkedRowIndex] = null;
+        if (checkedRowIndex === 0) {
+          prevSelectedRowsClone[0] = prevSelectedRowsClone[1];
+          prevSelectedRowsClone[1] = null;
+        }
+        return prevSelectedRowsClone;
+      }
+
+      if (prev.every((selectedVersion) => selectedVersion !== null)) {
+        const prevSelectedRowsClone = structuredClone(prev);
+        prevSelectedRowsClone[1] = flowVersion;
+        return prevSelectedRowsClone;
+      }
+
+      if (!prev[0]) {
+        return [flowVersion, null];
+      }
+      const flowB = prev[1] ?? flowVersion;
+
+      return [prev[0], flowB];
+    });
+  };
+
+  return (
+    <Box sx={tw`flex flex-col px-4 gap-y-6`}>
+      <InfoAlert
+        text={t.t(lang, (s) => s.components.flowVersions.info.compareInfo)}
+        localStorageKey="compareFlowVersions"
+        sxProps={tw`mb-4`}
+      />
+      <Table>
+        <TableBody>
+          {flow.versions
+            ?.sort(
+              (flowVersion, previous) =>
+                previous.versionID - flowVersion.versionID
+            )
+            .map((flowVersion) => (
+              <TableRowClick
+                key={`flowVersion_${flowVersion.id}v${flowVersion.versionID}`}
+                onClick={() => handleVersionSelection(flowVersion)}
+                sx={{
+                  backgroundColor: selectedVersions.some(
+                    (selectedVersion) =>
+                      selectedVersion?.id === flowVersion.id &&
+                      selectedVersion.versionID === flowVersion.versionID
+                  )
+                    ? tw`bg-unocha-primary bg-opacity-10`
+                    : undefined,
+                }}
+              >
+                <TableCell>
+                  {flowVersion.versionID === flow?.versionID && (
```

Comment:
According to its type, `flow` cannot be `null | undefined`
```suggestion
                  {flowVersion.versionID === flow.versionID && (
```